src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -281,19 +320,28 @@ private HttpRequestBase getHttpRequestBase(PrintStream logger, RequestAction req
         return httpRequestBase;
     }
 
-    private FilePath getOutputFilePath(AbstractBuild<?, ?> build) {
+    private FilePath getOutputFilePath(FilePath workspace, PrintStream logger) {
+        // No need to check for a null workspace, the workflow node step prevents workspace from being null.
         if (outputFile != null && !outputFile.isEmpty()) {
-            return build.getWorkspace().child(outputFile);
+            return workspace.child(outputFile);
         }
         return null;
     }
 
     private List<NameValuePair> createParameters(
-            AbstractBuild<?, ?> build, PrintStream logger,
+            Run<?, ?> run, PrintStream logger,
             EnvVars envVars) {
         if (!passBuildParameters) {
             return Collections.emptyList();
         }
+
+        // When executing as a workflow step, run is not an instance of AbstractBuild
+        if (!(run instanceof AbstractBuild<?, ?>)) {
+            // When executing as workflow, the user appends the parameters to the url as code, so just return an empty list
+            return Collections.emptyList();
+        }
```

Comment:
Another case where it might be clearer to make parameters be an argument to `performHttpRequest`, with `HttpRequestStep` simply passing `Collections.emptyList()`.